### DISCUSSION TITLE
AltServer-Linux... But it runs in the background

### DISCUSSION THREAD

**Discussion by TheWin9User** [powenn/AltServer-Linux-PyScript#22] [Ideas]
Something that would be amazing is that AltServer-Linux could run in the background so that the phone can automatically refresh apps without having to open the app multiple times. 

I say this because on windows, it would be running 24/7 and I would not need to worry about it. But with AltServer-Linux, everytime it sleeps, it resets/exits itself which can be unhelpful at times. 

So in summary, background implementation would be an amazing and useful addition to AltServer-Linux! 

Specially for people who cant always be on their pc but have it on.
Or is there already a way to run the app in the background?
Please let me know!

**Comment by sevenlayercookie** [comment]
I have achieved this simply by running the script at boot in a 'screen' that runs in the background. Easy enough to do by editing rc.local (which as a benefit, runs as root user, so no need to enter root password every time).

Make sure screen is installed first `sudo apt-get install screen`

Modify **/etc/rc.local** to include:

```
/usr/bin/screen -dm -S altserver python3 /path/to/main.py
```

Make sure it's included before the `exit 0` line.

To view the script while it's running, run `sudo screen -r`


My device has been refreshing successfully for a few days now with this.

**Comment by dollannn** [comment]
Wrote this quick service definition:

Ensure screen is installed: `sudo apt-get install screen`

In my case I have a root `/server/` that contains most of the stuff I run, adapt `/server/altserver/main.py` to where your `main.py` is located on your server.

`/etc/systemd/system/altserver.service`
```
[Unit]
Description=Altserver Python Daemon    
After=network.target

[Service]
ExecStart=/usr/bin/screen -dmS altserver /usr/bin/python3 /server/altserver/main.py
Type=forking
Restart=always

[Install]
WantedBy=multi-user.target
```

At last run `systemctl enable altserver` to run on boot & `systemctl start altserver` to start it up

List the screen process with screen -list and then screen -r **ID** to attach yourself to process to pair and install ipas etc.

**Reply by sevenlayercookie** [reply]
Nice!
